gamemode/modules/afk/sv_afk.lua

```diff
@@ -19,7 +19,10 @@ local function SetAFK(ply)
     local rpname = ply:getDarkRPVar("rpname")
     ply:setSelfDarkRPVar("AFK", not ply:getDarkRPVar("AFK"))
 
-    SendUserMessage("blackScreen", ply, ply:getDarkRPVar("AFK"))
+    if not ply.blackScreen then
+        ply.blackScreen = ply:getDarkRPVar("AFK")
+        SendUserMessage("blackScreen", ply, ply:getDarkRPVar("AFK"))
```

Comment:
You can re-use `ply.blackScreen` here:

```suggestion
        SendUserMessage("blackScreen", ply, ply.blackScreen)
```

Also, because of the condition this only sends the usermessage when a player is going AFK. Surely you'd want a message to turn it off when the player comes back from AFK?

Perhaps `ply.blackScreen == nil or ply.blackScreen ~= ply:getDarkRPVar("AFK")`?

Reply:
Good catch, dude.